Add Table of Contents and Glossary to Documentation (#829)
openblack/openblack: readme.md

[![Discord chat](https://img.shields.io/discord/608729286513262622?logo=discord&logoColor=white)](https://discord.gg/5QTexBU)
[![License](https://img.shields.io/github/license/openblack/openblack)](LICENSE.md)

> openblack is an open source reimplementation of [Black & White (2001)](https://en.wikipedia.org/wiki/Black_&_White_(video_game)) developed using modern C++ and rendering technologies like OpenGL and Vulkan.

Reviewer: ```suggestion
openblack is an open source reimplementation of [Black & White (2001)](https://en.wikipedia.org/wiki/Black_&_White_(video_game)) developed using modern C++ and rendering technologies such as Direct3D 11-12, Metal, OpenGL and Vulkan.
```

    <img src="https://user-images.githubusercontent.com/32263167/184559291-f37e44b3-c738-4c0f-82eb-28b7c84a3919.png" width="45%">
</a>

## Table of Contents
- [Glossary](#glossary)
- [Pre-built Binaries](#pre-built-binaries)
    - [Experimental](#experimental)
- [Building](#building)

>     - [Configuration for using vcpkg](#configuration-for-using-vcpkg-(recommended-for-new-users))

Reviewer: ```suggestion
    - [Configuration for using vcpkg](#configuration-for-using-vcpkg-recommended-for-new-users)
```

        - [Visual Studio Code](#visual-studio-code)
        - [Visual Studio 2022](#visual-studio-2022)
        - [Clion](#Clion)
        - [XCode](#xcode)
        - [Android Studio](#android-studio)
        - [Web Assembly](#web-assembly)
        - [Command Line](#command-line)

>     - [Configuration for using System Dependencies](#configuration-for-using-system-dependencies-(recommended-for-packagers))

Reviewer: ```suggestion
    - [Configuration for using System Dependencies](#configuration-for-using-system-dependencies-recommended-for-packagers)
```

        - [Arch Linux](#arch-linux)
    - [Contributing](#contributing)
- [Extracting Asset Files](#extracting-asset-files-from-original-game)
    - [Prerequisites](#prerequisites)
    - [Installation](#installation)
        - [On Windows](#on-windows)
        - [On Linux and macOS](#on-linux-and-macOS)
    - [Getting the Assets](#getting-the-assets)
        - [Using Them in Place](#using-them-in-place)
- [License](#license)

> ## Glossary

Reviewer: I guess I need convincing that the glossary is needed. Here's why.

> openblack
> Black & White (2001)

Firstly, this is explained in the first line. Second, if you don't know what these are, the project is not really for you.

> vcpkg, cmake, clang. msvc, xcode, submodules, git, presets, ninja

These are all concepts that matter for devs, if you're not a dev and not planning to compile yourself, you should be getting the pre-built binaries and not spending time reading the `Building` section. Especially git... this is github.

> Android Studio, Flatpak, AppImage, Snap, Arch Linux

I think those who look for these keywords will find them. These are all optional concepts and if you don't know what they are, then it's likely you don't need to interact with those instructions.

Maybe providing links to pages explaining the concepts (such as [Arch Linux](archlinux.org)) could help.

> - **OpenBlack**: An open-source reimplementation of *Black & White (2001)*.

Reviewer: ```suggestion
- **openblack**: An open-source reimplementation of *Black & White (2001)*.
```

## Configuration for using vcpkg (Recommended for new users)

> The easiest way to install all dependencies is via [vcpkg](https://github.com/Microsoft/vcpkg) which is comes with a [manifest file](https://github.com/microsoft/vcpkg/blob/master/docs/users/manifests.md).

Reviewer: ```suggestion
The easiest way to install all dependencies is via [vcpkg](https://github.com/Microsoft/vcpkg) which is configured with a [manifest file](https://github.com/microsoft/vcpkg/blob/master/docs/users/manifests.md).
```
